### DISCUSSION TITLE
Solidity Compiler Version

### DISCUSSION THREAD

**Discussion by Kuusho** [Cyfrin/foundry-full-course-cu#3132] [General]
You know how you can set a specific solidity compiler version in foundry.toml or some other ways.

Does anyone know a good reason or two why you would use a specific Solidity Compiler veresion either for a contract or as a test setting?

**Comment by EngrPips** [comment]
Every compiler version has one or more differences, and many of us don't know them because we have not been doing too much with the solidity language. Your contract will be compiled according to the set of rules of the compiler meaning the same contract might behave differently in one way or the other if compiled with different versions of the solidity compiler. Hence to be always sure of your contract behavior you set a specific compiler version.

**Reply by H-code9** [reply]
There are several reasons why using specific Solidity compiler versions, although they are not limited to these suggestions of mine.
1. Different versions have different security features (like how 0.8.0 added checks to prevent number overflows).
2. Some versions have bugs fixed while others might have issues, so you want to use a version you know is safe. A typical example is the Zksync testnet scenerio where we had to use version 0.8.24.
3. Each version might handle your code differently, so picking a specific version makes sure your contract works exactly as you expect.
4. even some may cost you more Gas than others.
I hope this answers your question.

**Reply by EngrPips** [reply]
Great Heads-up

**Reply by H-code9** [reply]
@EngrPips  Thanks sir.

**Reply by Kuusho** [reply]
Thanks you guys! That makes sense!

**Reply by EngrPips** [reply]
Indeed
